src/config/config.ts

```diff
@@ -325,29 +329,46 @@ export class Config implements IConfig {
   // eslint-disable-next-line default-param-last
   async runCommand<T = unknown>(id: string, argv: string[] = [], cachedCommand?: Command.Loadable): Promise<T> {
     debug('runCommand %s %o', id, argv)
-    const c = cachedCommand || this.findCommand(id)
+    let c = cachedCommand || this.findCommand(id)
     if (!c) {
       const matches = this.flexibleTaxonomy ? this.findMatches(id, argv) : []
       const hookResult = this.flexibleTaxonomy && matches.length > 0 ?
         await this.runHook('command_incomplete', {id, argv, matches}) :
         await this.runHook('command_not_found', {id, argv})
 
-      if (hookResult.successes[0]) {
-        const cmdResult = hookResult.successes[0].result
-        return cmdResult as T
-      }
+      if (hookResult.successes[0]) return hookResult.successes[0].result as T
+      if (hookResult.failures[0]) throw hookResult.failures[0].error
+      throw new CLIError(`command ${id} not found`)
+    }
 
-      if (hookResult.failures[0]) {
-        throw hookResult.failures[0].error
+    // console.log(this.pjson.oclif.jitPlugins)
+    if (Object.keys(this.pjson.oclif.jitPlugins ?? {}).includes(c.pluginName ?? '') && !this.plugins.find(p => p.name === c?.pluginName)) {
+      const pluginName = c.pluginName!
+      const pluginVersion = this.pjson.oclif.jitPlugins![pluginName]
+      const jitResult = await this.runHook('jit_plugin_not_installed', {
+        id,
+        argv,
+        command: c,
+        pluginName,
+        pluginVersion,
+      })
+      if (jitResult.failures[0]) throw jitResult.failures[0].error
+      if (jitResult.successes[0]) {
+        await this.loadPluginsAndCommands()
```

Comment:
I know this is only run once per jit but seems a bit expensive. Any other alternatives that would not be too difficult?

Reply:
Not that I've found. The alternative is to construct the plugin class for the newly added plugin, which is difficult because you have to know the file path it was installed at. It's easier to just reload the entire Config.

I'm not concerned about the performance hit. There's already a big cost to the user for the first run of a JIT plugin and reloading `Config` is minuscule compared to the time it takes to install the plugin.